llvm/lib/Object/XCOFFObjectFile.cpp

```diff
@@ -1288,7 +1288,7 @@ Expected<bool> XCOFFSymbolRef::isFunction() const {
   if (CsectAuxRef.getSymbolType() == XCOFF::XTY_LD)
     return true;
 
-  return false;
+  return createError("csect symbol has no valid symbol type.");
```

Comment:
Please follow the coding standards for error and warning messages here (https://llvm.org/docs/CodingStandards.html#error-and-warning-messages).

Also, grammatically, I'd expect it to be "csect symbol does not have a valid type" or something to that effect. Could you add more context, e.g. what the type actually is?

Reply:
The symbol type is for result of `CsectAuxRef.getSymbolType()`. For XCOFF symbols, there are four symbol types, see: 
```
enum SymbolType : uint8_t {
  XTY_ER = 0, ///< External reference.
  XTY_SD = 1, ///< Csect definition for initialized storage.
  XTY_LD = 2, ///< Label definition.
              ///< Defines an entry point to an initialized csect.
  XTY_CM = 3  ///< Common csect definition. For uninitialized storage.
};
```

So if the code goes here, the symbol type must be valid. I think `symbol type` means more than just `type` for XCOFF symbols.

Reply:
My request had three parts to it:
1) Following the coding standards.
2) Improve grammar.
3) Add more context.

1) and 2) are independent of 3) which is the one you've commented on.

Regarding 3) and the symbol type, what happens if the binary data contains a value that isn't present in the enum for the type field (for example)? Additional context could include, for example, that raw value (whatever it is).

Reply:
OK. Updated.